### PR TITLE
Support cloud hana / SPS 05+ versions

### PR DESCRIPTION
**Pull request by AndreasKlein**
@elakito Doing some research, I found querying SYS.OBJECTS being a valid way of determining the existence of `TABLE_PARTITIONS`.

### CODE REVIEW

File: src/main/scala/com/sap/kafka/connect/source/hana/HANASourceConnector.scala [hunk 1]
```diff
@@ -73,7 +75,22 @@ class HANASourceConnector extends SourceConnector {
...
+
+    checkStmtRs.next()
+    if (checkStmtRs.getInt(1) == 1) {
+      log.info(s"Found `TABLE_PARTITIONS` table to determine the partitions")
+      stmtToFetchPartitions += "PART_ID FROM SYS.TABLE_PARTITIONS WHERE "
+    } else {
+      log.info(s"Falling back to `M_CS_PARTITIONS` table to determine the partitions")
+      stmtToFetchPartitions += "PARTITION FROM SYS.M_CS_PARTITIONS WHERE "
+    }
+
```
Comment (elakito) [inline src/main/scala/com/sap/kafka/connect/source/hana/HANASourceConnector.scala:93]: I don't think this part with stmtToCheckPartitionTable is necessary. As table TABLE_PARTITIONS exists everywhere (even in older HANA SPS systems), stmtToCheckPartitionTable will always return 1 and we will be using TABLE_PARTITIONS anyway. In that case, we could actually replace line 76 directly with the appropriate query for TABLE_PARTITIONS.

So, basically we could use TABLE_PARTITIONS or M_CS_TABLES. The SPS 05 documentation mentions both TABLE_PARTITIONS and M_CS_TABLES as the replacement for M_CS_PARTITIONS. In contrast, the Cloud documentation mentions only TABLE_PARTITIONS as the replacement but M_CS_TABLES is available there. So, I asked the HANA colleagues if there is any particular reason for differentiating these two documentations and if there is any concern in using M_CS_TABLES. I hope to get the answer by tomorrow.

Comment (AndreasKlein) [reply src/main/scala/com/sap/kafka/connect/source/hana/HANASourceConnector.scala:93]: Thanks for the insights. Seems like only using `TABLE_PARTITIONS` is the way to go. I'll adjust the code accordingly. Do you think the fact that the cloud instance only has entries in `M_CS_TABLES` (for our tables with PART_ID=0), but not in `TABLE_PARTITIONS` matters? I wonder if that changes if a table is partitoined. Maybe we need to query multiple tables to get the right values?

Comment (elakito) [reply src/main/scala/com/sap/kafka/connect/source/hana/HANASourceConnector.scala:93]: Hi @AndreasKlein,  I got the information that the documentation of TABLE_PARTITIONS's PART_ID is incorrect regarding its describing PART_ID = 0 for non-partitioned tables. TABLE_PARTITIONS does not include non-partitioned tables. As for M_CS_TABLES, this table is used by many applications and it is safe to use it on both HANA 2.0 and Cloud HANA to get some simple partition information like the partition count.

So, I think we should use M_CS_TABLES. You mentioned in your issue ticket that you initially used this table in your code. 

Could you update your PR with using M_CS_TABLES? Thanks.

Comment (AndreasKlein) [reply src/main/scala/com/sap/kafka/connect/source/hana/HANASourceConnector.scala:93]: @elakito  done. A quick search showed that `M_CS_TABLES` seems to have been there in early SPS as well. We could drop the exists query and move away from using `M_CS_PARTITIONS` for all versions. Can you confirm that?

Comment (elakito) [reply src/main/scala/com/sap/kafka/connect/source/hana/HANASourceConnector.scala:93]: @AndreasKlein Yes. For M_CS_TABLES, you can expect to have this table everywhere, so you don't need the existence-query for this table.

Comment (AndreasKlein) [reply src/main/scala/com/sap/kafka/connect/source/hana/HANASourceConnector.scala:77]: @elakito done.

**Review by elakito — APPROVED** [review]
Looks good. Thank you for investigating this issue and provide this patch.

**Pull request merged**